packages/assets-controllers/src/Standards/NftStandards/ERC1155/ERC1155Standard.ts

```diff
@@ -130,6 +130,54 @@ export class ERC1155Standard {
     });
   };
 
+  /**
+   * Query for symbol for a given asset.
+   *
+   * @param address - ERC1155 asset contract address.
+   * @returns Promise resolving to the 'symbol'.
+   */
+  getAssetSymbol = async (address: string): Promise<string> => {
```

Comment:
Context note to reviewers.

`name()` and `symbol()` are not defined on ERC-1155.  The spec does not define a mechanism for the contract to report a collection name.  This has prevented showing a collection name during NFT approvals, which would be useful context.

The pattern we most commonly see implemented in contracts, supported by marketplaces, and contract templates, is that people are defining `name()` and `symbol()` anyway.

Reply:
We could pass tokenId as optional here (since its an optional arg to `getDetails` where this is called) and use the tokenId to query for the symbol and name from the JSON object returned by the uri?

Reply:
hey @adonesky1 
thank you for your comment.
the URI token stores information ( name, description, attributes ... ) for an NFT in the case of an erc1155.
e.g for tokenID 1: 
`
{
	"item_id": "1",
	"name": "Mini-Black Hole",
	"description": "High Goddess Cosmos in her infinite wisdom shaped these Mini-Black Holes to carefully siphon spiritual energy from existing planets to use in Celestial Conjunction to create new worlds. They take time to make - even Godlings cannot resist the all-consuming pull of black holes, and so it is Cosmos herself who must spin them out - but they are invaluable in purpose, effectively allowing any Godling to conduct the Celestial Conjunction and produce their own planets. \n\n The Mini-Black Hole NFT may be used in Celestial Conjunction with Apeiron Planet NFTs to \" breed \" new Planet NFTs. It may also be exchanged on the marketplace",
	"external_url": "http://apeironnft.com/",
	"image": "https://nftprops.apeironnft.com/godiversecollectionartwork/1.jpeg",
	"attributes": [
		{
			"trait_type": "Usage",
			"value": "Conjunction"
		}
	]
}
`

but in this case, we want to retrieve the contract level metadata ( more precisely name and symbol ), which is why it's not possible to use tokenURI.
this can be seen in the read part of the following [contract](https://polygonscan.com/address/0x482d4eE070aDBC35Aad4F4696f9aF3E11516Dc28#readContract) used to produce this bug

e.g: name() => return Apeiron Godiverse Collection
e.g: symbol() => return APEGC

Reply:
> but in this case, we want to retrieve the contract level metadata ( more precisely name and symbol ), which is why it's not possible to use tokenURI.

Yeah I'm not sure why we should consider these values to be restricted to contract level metadata. Also some ERC-1155 contracts store contract level metadata in tokenUri JSON blobs, just because of how the 1155 standard is written to exclude these values.

Reply:
most of ERC1155 contract are following the same patterns `name()` and `symbol()` that we have on this contract and they don't have this information on the `URI()` ( which is the case for this contract ).
this pattern is supported by marketplace like opensea , and we can see in this [link](https://opensea.io/collection/apeiron-godiverse-collection) the name and symbol are displayed correctly.
`tokenURI()` is not part of the ERC1155 standard , we use `URI()` instead.
so to resume `name()` and `symbol()` are not part of the standard but are commonly used and supported in popular marketplace

Reply:
also another thing, `tokenURI()` doesn't contain the contract level metadata according to the openSea standard , this information should be fetch from `contractURI()` or `name()` and `symbol()`

Reply:
Ok I'll relent on this for now. We definitely need to overhaul our handling of ERC1155s since there have been a bunch of developments in the past few years. OpenSea's standard is one of several we can try to follow but unfortunately there is no single standard on ERC1155 metadata so we shouldn't automatically privilege OpenSea's. But for now this is fine.

Reply:
If I'm not mistaken, the purpose of this PR is to reduce the number of ERC1155 nfts that appear without a name/symbol on the extension; even though the contract implements the name() and symbol().
This won't be the case for all ERC1155s; because it is not part of the standard; but today most of them implement them.

Eventually; if the NFT is 1155 and the contract does not implement those two functions it will fallback to the current behavior.

Reply:
I think checking `uri(tokenId)` for collection level metadata could eventually be useful too, in cases where a token id is known.  But `name()` and `symbol()` remain important both as fallbacks, and for  `setApprovalForAll` where there is no specific token id.